File: python_backend/app/domain/entities/payment_method.py
"""PaymentMethod domain entity (B5 — Billing).

PCI compliance (NFR-13): NEVER store PAN/CVV. Only Lemon Squeezy tokens +
display metadata (brand/last_four/exp). The actual card data lives in LS."""
from dataclasses import dataclass
from datetime import datetime
from typing import Literal, Optional

PaymentMethodType = Literal["card", "paypal", "other"]


@dataclass
class PaymentMethod:
    """A payment method stored as a token reference to Lemon Squeezy.

    NEVER includes raw card data — only display metadata."""

    id: str
    user_id: str
    lemon_squeezy_payment_method_id: Optional[str]
    type: PaymentMethodType
    brand: Optional[str]
    last_four: Optional[str]
    exp_month: Optional[int]
    exp_year: Optional[int]
    is_default: bool
    is_active: bool
    created_at: datetime

File: web/src/domain/entities/payment-method.ts
/**
 * PaymentMethod domain entity — TypeScript mirror of the backend
 * `PaymentMethod` entity.
 *
 * Backend contract (GET /api/billing/payment-methods as
 * `PaymentMethodResponse`):
 *
 *   {
 *     id: string,
 *     type: "card" | "paypal",
 *     brand: string | null,         // visa, mastercard, etc.
 *     last_four: string | null,
 *     exp_month: number | null,
 *     exp_year: number | null,
 *     is_default: boolean,
 *     is_active: boolean
 *   }
 *
 * NOTE: PCI compliance — the backend NEVER stores PAN/CVV. Only
 * Lemon Squeezy tokens + display metadata.
 */

export type PaymentMethodType = "card" | "paypal";

export interface PaymentMethod {
  id: string;
  type: PaymentMethodType;
  brand: string | null;
  lastFour: string | null;
  expMonth: number | null;
  expYear: number | null;
  isDefault: boolean;
  isActive: boolean;
}
